From 03540c066957d6647d164248a4b1c2c3b5fb1b8b Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:24:36 -0400
Subject: [tools] perf/completion: Use more comp words

Message-id: <1408462094-14194-108-git-send-email-jolsa@redhat.com>
Patchwork-id: 88079
O-Subject: [PATCH RHEL7.1 BZ1131394 107/325] perf completion: Use more comp words
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 6e0dc374a2c912a8a967ea8a4f9696dd4b0a6d3e
Author: Ramkumar Ramachandra <artagnon@gmail.com>
Date: Thu Jul 4 18:11:31 2013 +0530

description
===========
The completion words $words and $cword are available, so we might as
well use them instead of directly accessing COMP_WORDS.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/bash_completion b/tools/perf/bash_completion
index ee9c6d8..62e157d 100644
--- a/tools/perf/bash_completion
+++ b/tools/perf/bash_completion
@@ -92,15 +92,15 @@ __ltrim_colon_completions()
 type perf &>/dev/null &&
 _perf()
 {
- local cur prev cmd
+ local cur words cword prev cmd
 
  COMPREPLY=()
- _get_comp_words_by_ref -n : cur prev
+ _get_comp_words_by_ref -n =: cur words cword prev
 
- cmd=${COMP_WORDS[0]}
+ cmd=${words[0]}
 
  # List perf subcommands or long options
- if [ $COMP_CWORD -eq 1 ]; then
+ if [ $cword -eq 1 ]; then
   if [[ $cur == --* ]]; then
    COMPREPLY=( $( compgen -W '--help --version \
    --exec-path --html-path --paginate --no-pager \
@@ -110,13 +110,13 @@ _perf()
    COMPREPLY=( $( compgen -W '$cmds' -- "$cur" ) )
   fi
  # List possible events for -e option
- elif [[ $prev == "-e" && "${COMP_WORDS[1]}" == @(record|stat|top) ]]; then
+ elif [[ $prev == "-e" && "${words[1]}" == @(record|stat|top) ]]; then
   evts=$($cmd list --raw-dump)
   COMPREPLY=( $( compgen -W '$evts' -- "$cur" ) )
   __ltrim_colon_completions $cur
  # List long option names
  elif [[ $cur == --* ]];  then
-  subcmd=${COMP_WORDS[1]}
+  subcmd=${words[1]}
   opts=$($cmd $subcmd --list-opts)
   COMPREPLY=( $( compgen -W '$opts' -- "$cur" ) )
  fi
-- 
1.7.1